tuf/src/error.rs

```diff
@@ -41,11 +43,35 @@ pub enum Error {
         uri: String,
     },
 
+    /// An IO error occurred.
+    #[error("IO: {0}")]
```

Comment:
we've been not including source errors in the display text and then using anyhow to stringify the errors for logging, which itself includes the source errors in the stringification

not sure if that pattern would be useful here